wgpu-hal/src/metal/command.rs

```diff
@@ -903,15 +936,39 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
 
     // compute
 
-    unsafe fn begin_compute_pass(&mut self, desc: &crate::ComputePassDescriptor) {
+    unsafe fn begin_compute_pass(&mut self, desc: &crate::ComputePassDescriptor<super::Api>) {
         self.begin_pass();
 
         let raw = self.raw_cmd_buf.as_ref().unwrap();
+
         objc::rc::autoreleasepool(|| {
-            let encoder = raw.new_compute_command_encoder();
+            let descriptor = metal::ComputePassDescriptor::new();
+
+            let sba_descriptor = descriptor
+                .sample_buffer_attachments()
+                .object_at(0 as _) //TODO: move inside
+                .unwrap();
+            for (_i, at) in desc.timestamp_writes.iter().enumerate() {
+                //Problem here is that we can't attach the same counter sample buffer
+                //to the pass descriptor twice.
```

Comment:
Any ideas on a good solution for this? We could map from query_set -> sample buffer attachment index?

Reply:
I don't quite get the problem?